js/gallery.js

```diff
@@ -79,33 +86,51 @@ function galleryScroll(x){
     }, 1000);
 }
 
-function galleryControllerBar(c){
+function galleryControllerBar(control){
     $('.controls-bottom a').addClass('unselected');
     $('.controls-bottom a').removeClass('selected');
-    $(c).addClass('selected');
-    $(c).removeClass('unselected');
+    $(control).addClass('selected');
+    $(control).removeClass('unselected');
 }
 
 function galleryResize() {
     width = $(window).width();
     height = $(window).height();
     if($(window).width() > 767){
+        if(resizeChange){
+            current = galleryFirst();
+            galleryControllerBar($('.controls-bottom a[href="#item'+ (current) +'"]'));
+            resizeChange = false;
+        }
         $('.gallery').css('marginLeft', -(current - galleryFirst()) * width + 'px');
         $('.gallery-controls').css('marginLeft', (current - galleryFirst()) * width + 'px');
     }
     else {
         if(current == 1){
             current = galleryFirst();
             galleryControllerBar($('.controls-bottom a[href="#item'+ (current) +'"]'));
+            resizeChange = true;
         }
         $('.gallery').css('marginLeft', -(current - 2) * width + 'px');
         $('.gallery-controls').css('marginLeft', (current - 2) * width + 'px');
     }
 }
 
+function galleryAddControls(){
+    $('.controls-bottom').empty();
+    for(var i = 1; i < items + 1; i++){
+        if(i == 1){
+            $('.controls-bottom').append('<a href="#item'+ i +'" class="selected"><div></div></a>');
```

Comment:
I won't merge this kind of code. If you need templating, you shouldn't use jQuery for this, you need a templating engine.